OasisDEX/oasis-borrow
# [7641] open Aave borrow base form

# [open Aave borrow base form](https://app.shortcut.com/oazo-apps/story/7641/open-aave-borrow-base-form)
  
## User-facing changes

- open aave borrow positions at http://localhost:3000/borrow/aave/v2/open/wbtc?network=hardhat#simulate or http://localhost:3000/borrow/aave/v2/open/eth?network=hardhat#simulate
- only ETH and WBTC collaterals (at the two different URLs), and USDC debt for now
- user can set debt instead of risk ratio in the view
- user can open the position, but they can't view the position (it will probably still show as manage multiply position page)

## How to test 🧪

- ~~ensure test contracts deployed (https://www.notion.so/oazo/Local-development-5673b13353c34931b810281ab8e08783)~~ not necessary any more
- open a position using ETH and WBTC - all should work correctly on HH
- pretty thorough regression of steth/eth and multiply positions (open and manage)

## Tech notes:

- `openAaveStateMachine`
  - accepts a new parameter machine `openDepositBorrowAaveMachine` chooses which one to use based on the strategy config product type
  - has a new value on `context.userInput` - `debtAmount`, and associated event `SET_DEBT` to update it from the view
- `strategy` is now `IPositionTransition | ISimplePositionTransition` from the lib.  there are guards in `oasisActionsLibWrapper` called `transitionHasSwap` and `transitionHasMinConfigurableRiskRatio` that determine the difference between these types.  `ISimplePositionTransition` has no swap, and so the Oasis fee is set to zero in the UI when `ISimplePositionTransition` is detected.
- IStrategyConfig.adjustRiskView is now called `SecondaryInput`, has one interface for props and is implemented by both the risk slider and the debt input.
- Empty component here for the left hand side boxes for another story: features/borrow/aave/AaveBorrowManageComponent.tsx
- Depends on new version of lib from https://github.com/OasisDEX/oasis-earn-sc/pull/175
